xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -431,22 +431,22 @@ public Result selectConfig(PickSubchannelArgs args) {
           cluster =
               prefixedClusterSpecifierPluginName(action.namedClusterSpecifierPluginConfig().name());
           filters = selectedRoute.filterChoices.get(0);
+        } else {
+          throw new AssertionError();
         }
       } while (!retainCluster(cluster));
+      final RouteAction action = selectedRoute.routeAction;
```

Comment:
code readability nit: Consider a different name to differentiate  from `action` defined in the loop.

Reply:
Renamed to "routeAction"